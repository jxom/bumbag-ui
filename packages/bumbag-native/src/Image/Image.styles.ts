import { styled } from '../styled';
import { theme } from '../utils/theme';

export const StyledImage = styled.Image`
  ${theme('native.Image', 'styles.base')};
`;
