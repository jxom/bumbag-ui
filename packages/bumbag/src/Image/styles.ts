import { css, cssClass } from '../styled';
import { theme } from '../utils';

export const Image = (styleProps) => cssClass`
  ${
    styleProps.isFixed &&
    css`
      max-width: unset;

      ${theme(styleProps.themeKey, `styles.fixed`)(styleProps)};
    `
  }

  ${
    styleProps.fit === 'contain' &&
    css`
      object-fit: contain;

      ${styleProps.fitPosition &&
      css`
        object-position: ${styleProps.fitPosition};
      `};

      ${theme(styleProps.themeKey, `styles.contain`)(styleProps)};
    `
  }

  ${
    styleProps.fit === 'cover' &&
    css`
      object-fit: cover;

      ${styleProps.fitPosition &&
      css`
        object-position: ${styleProps.fitPosition};
      `};

      ${theme(styleProps.themeKey, `styles.cover`)(styleProps)};
    `
  }

  & {
    ${theme(styleProps.themeKey, `styles.base`)(styleProps)};
  }
`;
