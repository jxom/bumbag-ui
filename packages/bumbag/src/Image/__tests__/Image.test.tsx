import * as React from 'react';
import { renderHook } from '@testing-library/react-hooks';
import { Image } from '../Image';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(<Image src="test.png" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with CSS props', () => {
    const { container } = render(<Image alt="Bean" backgroundColor="red" src="test.png" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly for fixed image', () => {
    const { container } = render(<Image alt="Bean" isFixed src="test.png" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly for fit cover', () => {
    const { container } = render(<Image alt="Bean" fit="cover" src="test.png" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly for fit contain', () => {
    const { container } = render(<Image alt="Bean" fit="contain" src="test.png" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('theming', () => {
  it('Image.base should render correctly', () => {
    const { container } = render(<Image src="test.png" />, {
      theme: { Image: { styles: { base: { backgroundColor: 'red' } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Image.fixed should render correctly', () => {
    const { container } = render(<Image isFixed src="test.png" />, {
      theme: { Image: { styles: { fixed: { backgroundColor: 'red' } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Image.contain should render correctly', () => {
    const { container } = render(<Image fit="contain" src="test.png" />, {
      theme: { Image: { styles: { contain: { backgroundColor: 'red' } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Image.cover should render correctly', () => {
    const { container } = render(<Image fit="cover" src="test.png" />, {
      theme: { Image: { styles: { cover: { backgroundColor: 'red' } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly for className', () => {
    const { container } = render(<Image src="test.png" />, {
      theme: { Image: { defaultProps: { className: 'test', color: 'primary' } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
