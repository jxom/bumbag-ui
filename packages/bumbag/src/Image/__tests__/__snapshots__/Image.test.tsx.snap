// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defaultProps should render correctly for className 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-2 {
  color: #574feb !important;
}

<img
  class="emotion-0 emotion-1 emotion-2 test fp-Box fp-Image"
  src="test.png"
/>
`;

exports[`props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

<img
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Image"
  src="test.png"
/>
`;

exports[`props should render correctly for fit contain 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  object-fit: contain;
}

<img
  alt="Bean"
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Image"
  src="test.png"
/>
`;

exports[`props should render correctly for fit cover 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  object-fit: cover;
}

<img
  alt="Bean"
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Image"
  src="test.png"
/>
`;

exports[`props should render correctly for fixed image 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  max-width: unset;
}

<img
  alt="Bean"
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Image"
  src="test.png"
/>
`;

exports[`props should render correctly with CSS props 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-2 {
  background-color: #ff0000 !important;
}

<img
  alt="Bean"
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Image"
  src="test.png"
/>
`;

exports[`theming Image.base should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0.emotion-0 {
  background-color: #ff0000;
}

<img
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Image"
  src="test.png"
/>
`;

exports[`theming Image.contain should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  object-fit: contain;
}

.emotion-0.emotion-0 {
  background-color: #ff0000;
}

<img
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Image"
  src="test.png"
/>
`;

exports[`theming Image.cover should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  object-fit: cover;
}

.emotion-0.emotion-0 {
  background-color: #ff0000;
}

<img
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Image"
  src="test.png"
/>
`;

exports[`theming Image.fixed should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  max-width: unset;
}

.emotion-0.emotion-0 {
  background-color: #ff0000;
}

<img
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Image"
  src="test.png"
/>
`;
